packages/oasis3-mct/package.py

```diff
@@ -180,18 +197,140 @@ def __create_pkgconfig(self, spec, prefix):
             if nchars_written < len(text):
                 raise OSError
 
+    # https://spack-tutorial.readthedocs.io/en/ecp21/tutorial_advanced_packaging.html
+    @property
+    def libs(self):
+        # NOTE: The order matters during the linking step of the cice5 build!
+        # "-lpsmile.MPI1 -lmct -lmpeu -lscrip"
+        libraries = ["libpsmile.MPI1", "libmct", "libmpeu", "libscrip"]
+        return find_libraries(
+            libraries, root=self.prefix, shared=False, recursive=True
+        )
+
+    # Following pattern:
+    # https://spack-tutorial.readthedocs.io/en/latest/tutorial_buildsystems.html
+    def edit(self, spec, prefix):
+        SRCDIR = self.stage.source_path
+        makeinc_path = join_path(SRCDIR, self.__makeinc)
+        config = {}
+
+        # TODO: Clean up the debug / non-debug mechanism once the compiler
+        #       flag audit is finished.
+        #       https://github.com/ACCESS-NRI/ACCESS-OM/issues/12
+        NCI_INTEL_FLAGS = "-r8 -i4 -traceback -fpe0 -convert big_endian -fno-alias -ip -check noarg_temp_created"
```

Comment:
```suggestion
        NCI_INTEL_FLAGS = "-r8 -i4 -fpe0 -convert big_endian -fno-alias -ip -check noarg_temp_created"
```
I suggest reducing this to the options common to both variants and then adding the variant-specific flags later.

Reply:
Please comment in https://github.com/ACCESS-NRI/ACCESS-OM/issues/12

Reply:
I've made a comment in the above issue with respect to the flags being used, but this and the other comments above that concern layout don't depend on the flags used, they're about specifying information once: using a common base of options and adding to it rather than repeating the same information. 

I think it is better because it clearly delineates which options are **always** used, and which are dependent on optimisation/debugging/profiling etc.

Reply:
# TODO: Clean up the debug / non-debug mechanism once the compiler
    #       flag audit is finished.
    #       https://github.com/ACCESS-NRI/ACCESS-OM/issues/12